src/pch.h

```diff
@@ -4,3 +4,7 @@
 #include <plogr.h>
 
 using namespace Rcpp;
+
+#if MYSQL_VERSION_ID >= 80000 && MYSQL_VERSION_ID < 100000
```

Comment:
What does it mean if `MYSQL_VERSION_ID < 100000` ?

Reply:
So is this patch also valid when building against Connector-C for MariaDB?

Reply:
The patch does nothing except the change from `std::vector` to `boost::container::vector` if MariaDB is installed. MariaDB have not remove `my_bool` so far, and without `MYSQL_VERSION_ID < 100000`, the package cannot be built. But it is OK if the condition is added. I have tested that with MariaDB 10.1 and MariaDB 10.3 on Manjaro.

Reply:
But now I wonder if MariaDB would remove `my_bool` in the future. Maybe it is better to use `#if !defined(my_bool)` ?